### DISCUSSION TITLE
HLS to M2TS not working on 8.2.0.23/24

### DISCUSSION THREAD

**Discussion by shoooOny** [aclap-dev/video-downloadhelper#145] [Help and Q&A] [answered]
### Your question

FF 8.2.0.23 (stable) and I tried 8.2.0.23a1 (beta)
Chrome 8.2.0.24 (stable)

HLS to M2TS not working after update. Just saved it to ‘MP4’.
MP4 is very inconvenient because if it is interrupted while saving, the file is lost. Please solve this problem.

(After rolling back to version 8.1.4.0, the save function as M2TS worked normally again.)

**Comment by GrampaWildWilly** [comment]
My understanding is that this problem is not going to be resolved.  In fact, I believe the plan is to remove the setting from the product entirely.  It was something that Michel used for debugging something or other years ago & it just hung around in the product since then.  It seemed to cure some issues in previous versions of VDH but now, it is no longer needed.  You're probably going to get failures with HLS as M2TS enabled using current versions of VDH.  It's too bad about interrupted downloads.  That's just a fact of life.  I used to suffer terribly with them when I had an old, slow connection.  But I upgraded my ISP & things download faster than they can fail nowadays.  If interruptions are really a terrible problem for you, you might consider using another tool besides VDH.  VDH is not capable of restarting an interrupted download from the point of failure for HLS streams.  If you use the browser as your download processor, you can probably resume interrupted downloads in the browser.  But there are certain downloads that the browser can't handle.  That's why the CoApp exists.  So, as with so many choices in life, this one is a trade off.  You have to decide which way you're going to go.

**Reply by shoooOny** [reply]
I live in Korea and the internet speed is relatively fast, so even if the download stops midway, I can just try again, so that's okay.
However, if it is live streaming rather than a regular uploaded video, the story is different. If you have saved a live stream for hours but the download is interrupted for some reason, the file becomes unplayable and the entire file becomes useless.
In the case of M2TS saves, this function was essential because even if this problem occurred, the saved portion could be played.
However, it is a sad that this feature is disappearing. :(

**Reply by GrampaWildWilly** [reply]
It is my understanding that the mkv format does not suffer this same sort of problem when something is interrupted.  But if your content is not offered as mkv, you're still out of luck.  There is nothing that says you have to move up to the latest releases of VDH.  You could stay on a release that works for you.  You'll just have to hope you don't encounter a problem whose only cure is a new version of VDH.

**Reply by shoooOny** [reply]
I just hope that such a problem does not arise. Thank you for your kind and quick reply.

**Comment by Mattmjs** [comment]
The M2TS setting was necessary on the older companion 1x versions because it wasn't handling HLS streams properly.
On newer companion 2x versions which come package with ffmpeg, the M2TS setting has become redundant because ffmpeg solves
all those problems. With M2TS enable the audio was missing. I just tried with it enabled in 8.2.0.23 stable and it downloads as
mp4. It doesn't even come up as M2TS now. So it's basically obsolete now.

**Comment by paulrouget** [comment, marked as answer]
M2TS is not supported anymore.

But. One things I want to support is a M2TS-like format that would allow concatenating containers, visualize live streams as they are downloaded, and support channels (audio & video).

It's in my todo list.
